### PR TITLE
MESH-790/ Add RPC filtering by Link type

### PR DESCRIPTION
**Pull request by satyamakgec**

### CODE REVIEW

File: pallets/identity/src/lib.rs [hunk 1]
```diff
@@ -2049,6 +2049,77 @@ impl<T: Trait> Module<T> {
...
+                .filter(|link| {
+                    if !allow_expired {
+                        if let Some(expiry) = link.expiry {
+                            if expiry < now {
+                                return false;
+                            }
+                        }
+                    }
+                    match link.link_data {
+                        LinkData::DocumentOwned(..) => {
```
Comment (fmiguelgarcia) [inline pallets/identity/src/lib.rs:2075]: Try the following:
```
LinkData::DocumentOwned(..) => type_of_link == LinkType::DocumentOwnership,
LinkData::TickerOwned(..) => type_of_link == LinkType::TickerOwnership,
...
```

File: pallets/identity/src/lib.rs [hunk 2]
```diff
@@ -2049,6 +2049,77 @@ impl<T: Trait> Module<T> {
...
+                        LinkData::NoData => {
+                            if type_of_link == LinkType::NoData {
+                                true
+                            } else {
+                                false
+                            }
+                        }
+                    }
+                })
+                .map(|l| l)
```
Comment (fmiguelgarcia) [inline pallets/identity/src/lib.rs:2105]: Is this needed?

File: pallets/identity/src/lib.rs [hunk 3]
```diff
@@ -2049,6 +2049,77 @@ impl<T: Trait> Module<T> {
...
+                    if !allow_expired {
+                        if let Some(expiry) = l.expiry {
+                            if expiry < now {
+                                return false;
+                            }
+                        }
+                    }
+                    return true;
+                })
+                .map(|l| l)
```
Comment (fmiguelgarcia) [inline pallets/identity/src/lib.rs:2119]: Is this needed?

File: pallets/identity/src/lib.rs [hunk 4]
```diff
@@ -2049,6 +2049,77 @@ impl<T: Trait> Module<T> {
...
+    ///   with `allow_expired` boolean condition
+    pub fn get_filtered_links(
+        signatory: Signatory,
+        allow_expired: bool,
+        link_type: Option<LinkType>,
+    ) -> Vec<Link<T::Moment>> {
+        let now = <pallet_timestamp::Module<T>>::get();
+
+        if let Some(type_of_link) = link_type {
+            <Links<T>>::iter_prefix(signatory)
```
Comment (fmiguelgarcia) [inline pallets/identity/src/lib.rs:2065]: Unbounded operation, Is the number of links limited?

File: pallets/identity/src/lib.rs (hunk 2 above)
Comment (maxsam4) [reply pallets/identity/src/lib.rs:2105]: I am guessing Satyam is trying to take ownership of the element here. There are other ways to do it like cloning but this works as well :)

File: pallets/identity/src/lib.rs (hunk 4 above)
Comment (maxsam4) [reply pallets/identity/src/lib.rs:2065]: It's an "offchain" call so it's ok. We have plenty of unbounded operations in rpc.

**Comment by satyamakgec** [comment]
Tagged with WIP as there is some problem when accessing data from js.

**Comment by satyamakgec** [comment]
Problem solved ready to merge/re-review

**Pull request merged**